torchci/lib/drciUtils.ts

```diff
@@ -311,6 +312,13 @@ export async function hasSimilarFailures(
       authorEmail: record.authorEmail,
     };
 
+    const isEligibleCommit = await isEligibleCommitForSimilarFailureCheck(
+      failure.head_sha
+    );
```

Comment:
I feel like this could potentially be a lot of queries to rockset

Reply:
Yup, that's a valid concern.  I hope that this wouldn't stretch thing too much there.  IIRC, Rockset has caching, i.e. subsequent calls of the same query are cached.  Nevertheless, I could add a simple cache here to avoid checking the same SHA again and again